packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx

```diff
@@ -114,24 +114,26 @@ export class TooltipHost extends BaseComponent<ITooltipHostProps, ITooltipHostSt
       }
     }
 
-    this.setState({
-      isTooltipVisible: true
-    });
+    this.toggleTooltip(true);
   }
 
   // Hide Tooltip
   @autobind
   private _onTooltipMouseLeave(ev: any) {
-    this.setState({
-      isTooltipVisible: false
-    });
+    this.toggleTooltip(false);
   }
 
   // Hide Tooltip
   @autobind
   private _onTooltipCallOutDismiss() {
-    this.setState({
-      isTooltipVisible: false
-    });
+    this.toggleTooltip(false);
+  }
+
+  private toggleTooltip(isTooltipVisible: boolean) {
+    if (this.props.onTooltipToggle && this.state.isTooltipVisible !== isTooltipVisible) {
+      this.props.onTooltipToggle(isTooltipVisible);
+    }
```

Comment:
I thought on it, and I don't think so. We don't need render to happen, we safely deliver events in the order they happen.
Using setState callback would make up a more complex scenario, and won't affect the number and meaning of prop calls, so I'm keeping this simple (unless you see I'm missing some point).